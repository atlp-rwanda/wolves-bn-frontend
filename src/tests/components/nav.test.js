import React from 'react';
import configureMockStore from 'redux-mock-store';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import Nav from '../../components/nav/nav';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

describe('Nav component', () => {
  const store = mockStore({});

  store.dispatch = jest.fn();

  const component = renderer.create(
    <Router>
    <Provider store={store}>
      <Nav />
    </Provider>
    </Router>
  );
  it('should render nav with initial state', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});