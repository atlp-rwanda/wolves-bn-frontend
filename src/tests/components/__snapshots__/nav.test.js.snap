// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav component should render nav with initial state 1`] = `
<div
  className="nav"
>
  <h1>
     Logo
  </h1>
  <ul
    className="nav_links"
  >
    <a
      href="/home"
      onClick={[Function]}
    >
      <li>
        Home
      </li>
    </a>
    <a
      href="/profile"
      onClick={[Function]}
    >
      <li>
        Profile
      </li>
    </a>
    <a
      href="/login"
      onClick={[Function]}
    >
      <li>
        Login
      </li>
    </a>
  </ul>
</div>
`;
